consensus/service/src/byzantine_ledger/mod.rs

```diff
@@ -102,73 +106,75 @@ impl ByzantineLedger {
         // Helper function to broadcast an SCP message, as well as keep track of the highest
         // message we issued. This is necessary for the implementation of the
         // `get_highest_scp_message`.
-        let send_scp_message_ledger = ledger.clone();
-        let send_scp_message_broadcaster = broadcaster.clone();
-        let send_scp_message_node_id = node_id.clone();
-        let send_scp_message = move |scp_msg: Msg<TxHash>| {
-            // We do not expect failure to happen here since if we are attempting to send a
-            // consensus message for a given slot, we expect the previous block to exist (block not
-            // found is currently the only possible failure scenario for `from_scp_msg`).
-            let consensus_msg = ConsensusMsg::from_scp_msg(
-                &send_scp_message_ledger,
-                scp_msg.clone(),
-                msg_signer_key.as_ref(),
-            )
-            .unwrap_or_else(|_| panic!("failed creating consensus msg from {:?}", scp_msg));
 
-            // Broadcast the message to our peers.
-            {
-                let mut broadcaster = send_scp_message_broadcaster.lock().expect("mutex poisoned");
-                broadcaster.broadcast_consensus_msg(
-                    &send_scp_message_node_id.responder_id,
-                    &consensus_msg,
-                );
-            }
-
-            let mut inner = highest_outgoing_consensus_msg
-                .lock()
-                .expect("lock poisoned");
-            if let Some(highest_msg) = &*inner {
-                // Store message if it's for a newer slot, or newer topic.
-                // Node id (our local node) and quorum set (our local quorum set) are constant.
-                if consensus_msg.scp_msg.slot_index > highest_msg.scp_msg.slot_index
-                    || consensus_msg.scp_msg.topic > highest_msg.scp_msg.topic
+        let send_scp_message = {
```

Comment:
Moving the clones inside a scope also lets us give them short names.

Reply:
this seems like a really useful pattern!